homeassistant/components/synology_srm/device_tracker.py

```diff
@@ -71,14 +71,19 @@ def scan_devices(self):
         """Scan for new devices and return a list with found device IDs."""
         self._update_info()
 
-        return [device["mac"] for device in self.last_results]
+        return [device["mac"] for device in self.devices]
+
+    def get_extra_attributes(self, device) -> dict:
+        """Get the extra attributes of a device."""
+        filter_result = [result for result in self.devices if result["mac"] == device]
+        if filter_result:
+            return filter_result[0]
```

Comment:
What does this element contain? I think you need to filter out the various information as in the deleted line 97

Reply:
All of the additional properties as mentioned in the "After" section of the pull request... Should I allow some way to filter what additional attributes are required (CONF_MONITORED_CONDITIONS)?
Possibly with some defaults.. this is a router so thinking ip?

... Thoughts?

Reply:
IMHO filtering via monitored conditions is not useful for this case since you are only filtering attributes instead of setting up only the filtered sensors. 
Likewise see https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md on why monitored conditions is being deprecated. I think the reasons named apply here too.

Reply:
That article that you linked states:
> Integrations should expose all available data to the backend if that data is fetched in a single API request.

.. so does that mean it is good as it stands?.. or should I filter out some of it, as 97 used to?

Reply:
I think you should filter it to stay constistent with other device trackers.

Reply:
@springstan.. So what Information should I keep? ... I want most of it as I want to emulate a read-only view of Synology's Traffic control page with some additional information.

Reply:
Unfortunately, I cannot find any documentation to answer this question appropiately. However, since you are only adding attributes I think it is fine.

Reply:
@MartinHjelmare can you weigh in on this? Thanks 👍

Reply:
Well the complete list is in the bit of the OP as stated ... to me it almost seems a little pointless (to me) to extract the individual info for more or less everything and then adding it to the attribute ..

um... if I had to flick some...(crossed out):
band: 5G-1
~~connection: wifi~~ - can use is_wireless
current_rate: 433
dev_type: phone
hostname: i00-Phone
ip6_addr: '2406:3400:0419:4500:a247:51dd:a47c:a164'
ip_addr: 10.4.1.208
is_baned: false
is_beamforming_on: false
is_guest: false
~~is_high_qos: false~~
~~is_low_qos: false~~
~~is_manual_dev_type: false~~
~~is_manual_hostname: false~~
~~is_online: true~~ - effectively the state
is_parental_controled: false
is_qos: false
is_wireless: true
mac: 'c0:ee:fb:a2:12:a3'
~~max_rate: 433~~
mesh_node_id: 0
rate_quality: high
signalstrength: 76
transferRXRate: 208
transferTXRate: 20

... but that's just my preference ... someone may want to make use of some of the other data.

Reply:
Also ... I agree the documentation is a little lacking :)

Reply:
We want to set the attributes explicitly so we can see what they are and that the keys are formatted correctly. State attribute keys should be lowercase snakecase.

We can make a constant list with the attribute keys and then use it to iterate over while getting the corresponding values from the device data, to create the dict to return.

Reply:
Ahh that makes more sense ... didn't consider the snake case :) ... will get on to this tomorrow night since I am not home until then.

Reply:
Done ... but wasn't 100% sure if this should be done in the get_extra_attributes or the _update_info bit though so let me know if this needs changing.

Thanks.